lib/compat/wordpress-5.9/class-wp-theme-json-5-9.php

```diff
@@ -1889,7 +1889,7 @@ public static function get_from_editor_settings( $settings ) {
 			$theme_settings['settings']['color']['gradients'] = $settings['gradients'];
 		}
 
-		if ( isset( $settings['fontSizes'] ) ) {
```

Comment:
Hi, I've looked at [this code in 5.8](https://github.com/WordPress/wordpress-develop/blob/5.8/src/wp-includes/class-wp-theme-json.php#L1205) and nothing has changed. It sounds like the way data is passed to `get_from_editor_settings( $settings )` has changed and we no longer make sure it's an array (empty or otherwise). I'd love it if we could trace what happened and provide a fix at the root of the issue. Otherwise, we may run into the same issue in other places.

Reply:
Thanks for taking a look!

The root cause is perhaps happening in [wp-includes/block-editor.php](https://github.com/WordPress/wordpress-develop/blob/5.9/src/wp-includes/block-editor.php#L231):

```
$font_sizes = current( (array) get_theme_support( 'editor-font-sizes' ) );
if ( false !== $font_sizes ) {
  $editor_settings['fontSizes'] = $font_sizes;
}
```

- get_theme_support() returns  [`$_wp_theme_features[ $feature ]`](https://developer.wordpress.org/reference/functions/get_theme_support/) by default when no $args is passed
- this evaluates to ["editor-font-sizes"]=> bool(true)
- when the value for $font_sizes is assigned:
  - it's cast to an array type
  - `current()` returns a value of `1`
- this makes `false !== $font_sizes` always return as true
- the value of $editor_settings['fontSizes'] is then set as `1` and is passed along in `$settings`

I notice this was [the same in 5.8](https://github.com/WordPress/wordpress-develop/blob/7f4efc55b682e3f6b8e8afeb80cbd967cc4a4a14/src/wp-includes/block-editor.php#L215) so I'm now wondering if I'm mistaken about the warning not showing prior to 5.9. I recently inherited a bunch of sites and was running updates on those, so quite possible I got confused — will test with 5.8 when I next get a moment.

Reply:
Testing in 5.8:
- no warning with a classic theme
- but there is a warning with a block theme + Gutenberg

@oandregal am I correct thinking this should be addressed in core via `wp-includes/block-editor.php`.I'll open a ticket there, if so.